proof/crefine/ARM_HYP/Finalise_C.thy

```diff
@@ -2179,29 +2179,38 @@ lemma associateVCPUTCB_ccorres:
           apply ceqv
          apply (rule ccorres_move_c_guard_vcpu)
          apply clarsimp
-         apply (subst ccorres_seq_skip'[symmetric])
          apply (rule ccorres_split_nothrow[where r'=dc and xf'=xfdc])
-             apply (rule  setObject_vcpuTCB_updated_Basic_ccorres[where tptr="Some tptr" and t="Map.empty"
-                                                                 , simplified option_to_ctcb_ptr_def, simplified])
+             apply (rule setObject_vcpuTCB_updated_Basic_ccorres[where tptr="Some tptr" and t="Map.empty",
+                                                                 simplified option_to_ctcb_ptr_def, simplified])
             apply ceqv
-           apply (rule ccorres_return_Skip)
-          apply (rule wp_post_taut)
+           apply (rule ccorres_pre_getCurThread, rename_tac curThread)
+           apply (subst ccorres_seq_skip'[symmetric])
+           apply (rule ccorres_split_nothrow[where r'=dc and xf'=xfdc])
+               apply (rule_tac R="\<lambda>s. curThread = ksCurThread s" in ccorres_when)
+                apply (intro allI impI)
+                apply (unfold mem_simps)[1]
+                apply (clarsimp simp: rf_sr_ksCurThread)
+               apply (ctac add: vcpu_switch_ccorres_Some)
+              apply ceqv
+             apply (rule ccorres_return_Skip)
+            apply (rule wp_post_taut)
+           apply (vcg exspec=vcpu_switch_modifies)
+          apply (rule_tac Q="\<lambda>_. pspace_aligned' and pspace_distinct' and valid_objs' and no_0_obj'
+                                 and valid_arch_state' and vcpu_at' vcpuptr"
+                 in hoare_post_imp, fastforce)
```

Comment:
indent `in` by two as it falls under `rule_tac`

Reply:
also, there seems a fair bit of hoare_post_imp added in this proof; can you say why wp couldn't deal with it (with some assistance)?

Reply:
Oh man, that (indenting to first param) is actually not bad either. I support doing that unless you're really running low on horizontal space in the proof. @lsf37 @corlewis  : agree?

Reply:
I think you've gotten to the core of the different viewpoints, I've definitely been considering the `in` as belonging to the `rule_tac`.